apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -71,9 +77,26 @@ func (host *BareMetalHost) validateChanges(old *BareMetalHost) []error {
 		errs = append(errs, fmt.Errorf("bootMACAddress can not be changed once it is set"))
 	}
 
+	if err := validateAddStatusAnnotation(old, host); err != nil {
+		errs = append(errs, err)
+	}
+
 	return errs
 }
 
+func validateAddStatusAnnotation(old *BareMetalHost, new *BareMetalHost) error {
+
+	oldStatusNotEmpty := (old.Status != (BareMetalHostStatus{}))
+	newAnnotationNotEmpty := (new.Annotations[StatusAnnotation] != "")
+	newAnnotationIsDifferent := (new.Annotations[StatusAnnotation] != old.Annotations[StatusAnnotation])
+
+	if oldStatusNotEmpty && newAnnotationNotEmpty && newAnnotationIsDifferent {
+		return fmt.Errorf("cannot add new statusannotation when status-subresource is already present")
```

Comment:
instead of literal "statusannotation" you probably meant to insert the value of `StatusAnnotation`.

Reply:
@dtantsur do you mean, we should return the values of `New.StatusAnnotation` and `Old.StatusAnnotation` and state that they are not equal ?

Reply:
He means put `cannot add new "baremetalhost.metal3.io/status" annotation`